src/components/solvers/TextInputMask.tsx

```diff
@@ -46,14 +46,14 @@ export const TextInputMask = (props: TextInputMaskProperties) => {
                           value={text}
                           minHeight="10rem"
                           isInvalid={errorString != ""}
-                          onChange={onTextAreaChange}/>
+                          onChange={x => onTextChanged(x.target.value)}/>
 
                 <Text backgroundColor="tomato">{errorString}</Text>
 
                 <InputButtonPanel
                     helpBody={<Help/>}
                     problemString={text}
-                    setProblemString={setText}
+                    setProblemString={x => onTextChanged(x as string)}
```

Comment:
One thing I wasn't sure about is how to handle this properly.
`x` is a `React.SetStateAction<string>`, but actually just a string that we can pass to `onTextChanged`.
I added the part `as string` to avoid IDE warnings.